compiler/compile-config/compile-config-2.01-grounding

```diff
@@ -427,16 +375,27 @@ def withRedefinitionsWithJoinsForIdColumns:
 
             # materialize factors using user input_query that pulls in assigned ids to involved variables
             deepdive create table \(.factorsTable | @sh) as \(
-                # allow .input_query to refer to the assigned variable ids by redefining the variable table names with SQL CTE
+                # Not using DISTINCT: User decides if they want to have duplicate factors
```

Comment:
Note that `factorsTable` can be a view.  Materializing this can happen in the `factors/*/dump` process.  weightsTable below can come directly from the `input_query` instead.  Change can be simple as `deepdive create view ... as ...` or we could move and plug the query into the dump process.

In general, only the ones up for sequential id assignment have to be materialized as table.  If user's `input_query` is costly we could materialize that instead, but I think factor input_query is expected to generate bloat by design since the result is supposed to be projected and group'ed/distinct'ed, so maybe that's a bad idea.